### PR TITLE
Feature: add slideshow option to gallery

### PR DESCRIPTION
**Pull request by andi34**
- [x] make slideshow time adjustable via admin panel

Who likes to give it a try can use my fork at https://github.com/andi34/photobooth which also contains different other open pull-requests and some other extras.

Todo:
- [ ] add auto-refresh `$imagelist` array and make time adjustable via admin panel
  Question/to think about: if slideshow is running, open first image from `$imagelist` and start slideshow again after reloading.
  Any idea how to make this possible?

Initial work taken from https://github.com/dimsemenov/PhotoSwipe/issues/753#issuecomment-225374602

### CODE REVIEW

File: config/config.inc.php [hunk 1]
```diff
@@ -69,6 +69,7 @@
 $config['scrollbar'] = false;
 $config['show_date'] = false; // only works if file_format_date = true
 $config['gallery']['date_format'] = 'd.m.Y - G:i';
+$config['slideshow_pictureTime'] = '3000';
```
Comment (sualko) [inline config/config.inc.php]: Why is this a string?

File: resources/js/photoinit.js [hunk 2]
```diff
@@ -196,6 +209,32 @@ function initPhotoSwipeFromDOM (gallerySelector) {
...
+        }
+        const title = running ? 'Pause Slideshow' : 'Play Slideshow';
+        $(el).removeClass(running ? 'play' : 'pause')
+            .addClass(running ? 'pause' : 'play')
+            .prop('title', title);
+        ssRunning = running;
+    }
+
+    function gotoNextSlide() {
+        if (ssRunning && Boolean(gallery)) {
```
Comment (sualko) [inline resources/js/photoinit.js]: Why is the `Boolean` constructor used at this point?

File: resources/js/photoinit.js [hunk 3]
```diff
@@ -2,7 +2,12 @@
 /* global photoBooth */
 function initPhotoSwipeFromDOM (gallerySelector) {
 
-    let gallery;
+    let gallery,
+        ssRunning = false,
+        ssOnce = false;
```
Comment (sualko) [inline resources/js/photoinit.js]: What is `ssOnce` doing?

File: resources/js/photoinit.js [hunk 4]
```diff
@@ -2,7 +2,12 @@
 /* global photoBooth */
 function initPhotoSwipeFromDOM (gallerySelector) {
 
-    let gallery;
+    let gallery,
+        ssRunning = false,
```
Comment (sualko) [inline resources/js/photoinit.js]: I personally don't like var listing, but that's a personal thing.

File: resources/js/photoinit.js [hunk 5]
```diff
@@ -196,6 +209,32 @@ function initPhotoSwipeFromDOM (gallerySelector) {
...
+    $(ssButtonClass).on('click touchstart', function(e) {
+        e.preventDefault();
+        e.stopPropagation();
+        // toggle slideshow on/off
+        setSlideshowState(this, !ssRunning);
+    });
+
+    function setSlideshowState(el, running) {
+        if (running) {
+            setTimeout(gotoNextSlide, ssDelay / 2.0);
```
Comment (sualko) [inline resources/js/photoinit.js]: Why do we go to the next slide after half the time?

File: resources/js/photoinit.js (hunk 2 above)
Comment (andi34) [reply resources/js/photoinit.js]: `yarn eslint` told so....

File: resources/js/photoinit.js (hunk 5 above)
Comment (andi34) [reply resources/js/photoinit.js]: https://github.com/dimsemenov/PhotoSwipe/issues/753#issuecomment-225374602

File: config/config.inc.php (hunk 1 above)
Comment (andi34) [reply config/config.inc.php]: Because of the `''` ? We've done that on all other config options too, added for consistency

**Comment by andi34** [comment]
Branch exists here @sualko . You'd like to push your optimisations here?

File: config/config.inc.php (hunk 1 above)
Comment (sualko) [reply config/config.inc.php]: The picture quality is just a number, but you are right, most of the numerical values are strings. Usually this is no problem, just in js could be some misbehavior.

File: resources/js/photoinit.js (hunk 2 above)
Comment (sualko) [reply resources/js/photoinit.js]: Never saw this before and I don't get the advantage, but it's also no big deal.

**Comment by sualko** [comment]
I have currently no time to contribute code to this project, but I can review and comment on stuff if you like. Sorry. It's currently a busy time.

**Comment by andi34** [comment]
Don't worry, take your time. I've no idea how to finish this...

**Comment by andi34** [comment]
mh... any help on how to refresh `$images` every X seconds?
i tried
```
-$images = getImagesFromDB();
+function getImages() {
+       $images = getImagesFromDB();
+}
+
+while (true)
+{
+   getImages(); // Call your function
+   sleep(30);   //wait for 30 sec for next function call
+}
```
but doesn't seems to work... page is loading forever

**Comment by andi34** [comment]
In the meantime there have been some adjustments made. All sources can be found here: https://github.com/andi34/photobooth

**Pull request closed without merging**